packages/drivers/odsp-driver/src/fetcher.ts

```diff
@@ -0,0 +1,92 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { IsoBuffer } from "@fluidframework/common-utils";
+import { OdspErrorType } from "./odspError";
+import { fetchAndParseAsBufferHelper, fetchAndParseAsJSONHelper, IOdspResponse } from "./odspUtils";
+
+export class Fetcher {
+    constructor(
+        private fluidEpoch: string | undefined,
+        private readonly logger?: ITelemetryLogger) {
+    }
+
+    public async fetchAndParseAsJSON<T>(
+        url: string,
+        fetchOptions: {[index: string]: any},
+        addInBody: boolean = false,
+    ): Promise<IOdspResponse<T>> {
+        // Add epoch either in header or in body.
+        this.addEpochInRequest(fetchOptions, addInBody);
+        let response: IOdspResponse<T>;
+        try {
+            response = await fetchAndParseAsJSONHelper<T>(url, fetchOptions);
+            this.extractEpochFromResponse(response.headers);
+        } catch (error) {
+            this.checkForEpochError(error);
+            throw error;
+        }
+        return response;
+    }
+
+    public async fetchAndParseAsBuffer<T>(
+        url: string,
+        fetchOptions: {[index: string]: any},
+        addInBody: boolean = false,
+    ): Promise<IOdspResponse<IsoBuffer>> {
+        // Add epoch either in header or in body.
+        this.addEpochInRequest(fetchOptions, addInBody);
+        let response: IOdspResponse<IsoBuffer>;
```

Comment:
You can make this (and other) functions shorter by one line :)

this.addEpochInRequest(fetchOptions, addInBody);
 try {
            const response = await fetchAndParseAsBufferHelper(url, fetchOptions);
            this.extractEpochFromResponse(response.headers);
            return response;
        } catch (error) {
            this.checkForEpochError(error);
            throw error;
        }

Reply:
done